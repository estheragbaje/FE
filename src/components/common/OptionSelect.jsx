import React from 'react';
import PropTypes from 'prop-types';

const OptionSelect = ({
  name, value, label, options, className, onChange,
}) => (
  <>
    <label htmlFor="type" className="pb-3">{label}</label>
    <select name={name} value={value} className={className} onChange={onChange}>
      <option value="" disabled>
        Choose option
      </option>
      {options.map((option) => (
        <option key={option.value} value={option.value}>
          {option.name}
        </option>
      ))}
    </select>
  </>
);

OptionSelect.defaultProps = { options: [] };

OptionSelect.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.object),
};

export default OptionSelect;
