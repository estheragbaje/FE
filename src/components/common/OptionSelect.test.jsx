import React from 'react';
import { shallow } from 'enzyme';

import OptionSelect from './OptionSelect';

describe('<OptionSelect />', () => {
  it('renders the OptionSelect component correctly', () => {
    const wrapper = shallow(<OptionSelect />);
    expect(wrapper).toMatchSnapshot();
  });
});
