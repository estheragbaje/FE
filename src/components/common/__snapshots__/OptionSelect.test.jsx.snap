// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionSelect /> renders the OptionSelect component correctly 1`] = `
<Fragment>
  <label
    className="pb-3"
    htmlFor="type"
  />
  <select>
    <option
      disabled={true}
      value=""
    >
      Choose option
    </option>
  </select>
</Fragment>
`;
